.cbmc-batch/jobs/aws_byte_buf_from_c_str/aws_byte_buf_from_c_str_harness.c

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/common/byte_buf.h>
+#include <proof_helpers/make_common_data_structures.h>
+
+void aws_byte_buf_from_c_str_harness() {
+    size_t len;
+    char *c_str;
+
+    if (nondet_bool()) {
+        c_str = NULL;
+    } else {
+        /* Assume the length of the string is bounded by some max length */
+        __CPROVER_assume(len <= MAX_BUFFER_SIZE);
```

Comment:
There is a function for this https://github.com/awslabs/aws-c-common/blob/master/.cbmc-batch/include/proof_helpers/make_common_data_structures.h#L135